src/TempleLineOfCredit.sol

```diff
@@ -0,0 +1,318 @@
+            revert ExceededCollateralAmonut(msg.sender, collateralAmount, withdrawalAmount);
+        }
+
+        uint256 borrowCapacity = _maxBorrowCapacity(collateralAmount - withdrawalAmount);
+        
+        if (positions[msg.sender].debtAmount > borrowCapacity ) {
+            revert WillUnderCollaterlize(msg.sender, withdrawalAmount);
+        }
+
+        positions[msg.sender].collateralAmount -= withdrawalAmount;
+        collateralToken.safeTransfer(
+            msg.sender,
+            withdrawalAmount
+        );
+
+        emit Withdraw(msg.sender, withdrawalAmount);
+    }
+
+   /**
+     * @dev Allows borrower to repay borrowed amount
+     * @param repayAmount is the amount to repay
+     */
+    function repay(uint256 repayAmount) external {
+        if (repayAmount == 0) revert InvalidAmount(repayAmount);
+        positions[msg.sender].debtAmount = getTotalDebtAmount(msg.sender);
+
+        if (repayAmount >  positions[msg.sender].debtAmount) {
+            revert ExceededBorrowedAmount(msg.sender, positions[msg.sender].debtAmount, repayAmount);
+        }
+
+        positions[msg.sender].debtAmount -= repayAmount;
+        debtBalance += repayAmount;
+        
+        // If more than 1 interest rate period has passed update the start-time
+        if (block.timestamp - positions[msg.sender].createdAt >= interestRatePeriod || positions[msg.sender].createdAt == 0  ) {
+            positions[msg.sender].createdAt = block.timestamp;
+        }
+        debtToken.safeTransferFrom(
+            msg.sender,
+            address(this),
+            repayAmount 
+        );
+        emit Repay(msg.sender, repayAmount);
+    }
+   
+   /**
+     * @dev Allows operator to liquidate debtors position
+     * @param debtor the account to liquidate
+     */
+    function liquidate(address debtor) external onlyOperators {
+
+        if (getCurrentCollaterilizationRatio(debtor) >= minCollateralizationRatio) {
+            revert OverCollaterilized(debtor);
+        }
+
+        uint256 totalDebtOwed = getTotalDebtAmount(debtor);
+        uint256 collateralSeized = (totalDebtOwed * debtPrice) / collateralPrice;
+
+        if (collateralSeized > positions[debtor].collateralAmount) {
+            collateralSeized = positions[debtor].collateralAmount;
+        }
+
+        positions[debtor].collateralAmount -= collateralSeized;
+        positions[debtor].debtAmount = 0;
+        positions[debtor].createdAt = 0;
+        collateralToken.safeTransfer(
+            debtCollector,
+            collateralSeized 
+        );
+
+        emit Liquidated(debtor, totalDebtOwed, collateralSeized);
+
+    }
+
+    function getCurrentCollaterilizationRatio(address account) public view returns(uint256) {
+        if (positions[account].debtAmount == 0) {
+            return 0;
+        } else {
+            return ((positions[account].collateralAmount * collateralPrice * 100) / getTotalDebtAmount(account) / debtPrice);
+        }
+    }
+
```

Comment:
Should we add a `recoverToken()` function which can recover:
1. any non-debt token?
2. any debt tokens > `debtBalance` (ie someone sent the tokens without calling `removeDebt()`)